### PR TITLE
Split Source into Source/SourceActor, and Breakpoint into Breakpoint/…

### PR DESCRIPTION
**Pull request by bhackett1024**
### Summary of Changes

* Add a SourceActor object representing a SourceClient connected to a source in a thread.
* Change Source to be independent of threads, and provide a mapping Source -> SourceActor[] in redux state.
* Add a BreakpointActor object representing a BreakpointClient, and which is associated with a SourceActor instead of a Source (as Breakpoint is).

This has the general effect of making sources and breakpoints in the client thread-independent, as was the case before windowless worker debugging landed.  More background on this change can be found in bug 1518884.

### Test Plan

Passes unit tests.
Manually tested on TodoMVC.

There are still some holes in the implementation wrt threads.  Mainly, as sources from new threads appear, BreakpointActors will not be created if there are applicable sites for existing breakpoints in those sources.  I'd like to fix this in followup, as there are a fair number of cleanups to client breakpoint state that can be performed once this lands.

### CODE REVIEW

File: src/utils/breakpoint/index.js [hunk 1]
```diff
@@ -45,6 +47,8 @@ export function locationMoved(
   );
 }
 
+// The ID for a Breakpoint is derived from its location in its Source.
+// FIXME rename this to makeBreakpointId.
```
Comment (darkwing) [inline src/utils/breakpoint/index.js]: Just calling this FIXME out.

**Comment by jasonLaster** [comment]
I'm seeing this when i have a breakpoint in todo-view.js#34 and reload

screen shot 2019-01-25 at 12 31 57 pm

File: src/actions/breakpoints/syncBreakpoint.js [hunk 2]
```diff
@@ -121,18 +120,32 @@ export async function syncBreakpointPromise(
...
-    const id = pendingBreakpoint.disabled ? "" : existingClient.id;
+  if (pendingBreakpoint.disabled || isSameLocation) {
+    // Make sure the breakpoint is installed on all source actors.
+    if (!pendingBreakpoint.disabled) {
+      for (const sourceActor of sourceActors) {
+        const sourceActorLocation = makeSourceActorLocation(
+          sourceActor,
+          generatedLocation
+        );
+        if (!client.getBreakpointByLocation(sourceActorLocation)) {
```
Comment (jasonLaster) [inline src/actions/breakpoints/syncBreakpoint.js]: why is this needed?
